### PR TITLE
[ENG-2497] Remove validated-model-forms from developer-apps pages

### PR DESCRIPTION
**Pull request by futa-ikeda**
- Ticket: [ENG-2497]
- Feature flag: n/a

## Purpose
- Phase out validated-model-forms as a way of validating forms and instead using form-controls and changesets for the settings.applications route

## Summary of Changes
- Create a new component that is shared between the developer-apps/create and developer-apps/edit pages
- Removing validated-model-forms from these forms and replacing them with form-controls

## Side Effects
- I think there is a small side-effect with how validation errors go away. When a user hits the `create`/`save` button and tries to submit an invalid form (e.g. they try to create/edit an app with an invalid callbackUrl), validation errors will show. When they go to edit the callbackUrl by typing anything into that text field, the validation error goes away, whereas before it would stay until the user clicks the `create` or `save` button. It's not a major issue, but I think it's a small deviation from before.

https://user-images.githubusercontent.com/51409893/103942860-0ecb0200-50ff-11eb-91c7-2303a210275f.mov

## QA Notes
- A quick look at the developer-apps page would be appreciated. None of the functionality should be different (aside from the small deviation with regards to the validation errors stated above)

[ENG-2497]: https://openscience.atlassian.net/browse/ENG-2497

### CODE REVIEW

File: app/settings/developer-apps/-components/app-form/component.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+@tagName('')
+export default class DeveloperAppForm extends Component {
+    @service intl!: Intl;
+    @service router!: RouterService;
+    @service toast!: Toast;
+
+    // Required arguments
+    developerApp?: DeveloperApp;
+    appTaskInstance?: TaskInstance<DeveloperApp>;
+    mode!: string;
```
Comment (futa-ikeda) [inline app/settings/developer-apps/-components/app-form/component.ts:26]: I don't really like this string method of toggling what is shown on the template, so any suggestions here would be welcome

Comment (brianjgeiger) [reply app/settings/developer-apps/-components/app-form/component.ts:26]: If it's just toggling between create mode or not, you could make a createMode boolean to set if `mode == 'create'`, and use that for the template logic.

Comment (futa-ikeda) [reply app/settings/developer-apps/-components/app-form/component.ts]: It seems so obvious now that is should just be a boolean...

**Review by adlius — DISMISSED** [review]
Functionality tested locally and works fine.

File: app/settings/developer-apps/create/controller.ts [hunk 2]
```diff
@@ -1,23 +1,21 @@
...
     @service toast!: Toast;
+    developerApp!: DeveloperAppModel;
 
-    @action
-    onSave(developerApp: DeveloperApp & ChangesetDef) {
-        this.toast.success(this.intl.t('settings.developer-apps.created'));
-        this.router.transitionTo('settings.developer-apps.edit', developerApp.get('id'));
+    init() {
+        super.init();
+        this.developerApp = this.store.createRecord('developer-app');
```
Comment (fabmiz) [inline app/settings/developer-apps/create/controller.ts:19]: I am thinking `setUpController` may be a better fit for this use case. My understanding is Controller `init()` gets called once (only one instance of the controller exists in the application) and so next time this route is visited the init() does not run i.e `this.developerApp` is still same instance as last (might lead to errors within the store/ ember-data). `setUpController` runs every time the route is entered (allowing us to create a new instance of the `developer-app` on every visit.

**Review by fabmiz — APPROVED** [review]
💯

**Pull request merged**